### PR TITLE
feat(tracking): Add tracking on slice machine start

### PR DESCRIPTION
**Pull request by xrutayisire**
## Context

- Completes DT-1983

## To Do

- Test for new manager function to detect git provider

## The Solution

- Added a new manager function to detect git provider 
- Added a new event type for slice machine start
- Added all the properties that can be useful for us in order to see how our users use our product
- Fixed a bug that was adding `_includeEnvironmentKind` to the event payload

## Impact / Dependencies

- None, the start process is not longueur as it's done after

### CODE REVIEW

File: packages/manager/src/managers/git/GitManager.ts
Comment (angeloashmore) [inline packages/manager/src/managers/git/GitManager.ts:475]: We use `"gitHub" | "gitLab" | "bitbucket"` as provider identifiers, which are the camelcase versions of each provider's name.

https://github.com/prismicio/slice-machine/blob/9388a4b033a69b61abb4b57e6831abf722237519/packages/manager/src/constants/GIT_PROVIDER.ts#L5

We only support GitHub today, but we plan to support GitLab and Bitbucket in the future.

We'll probably want consistency in a general purpose method like this one. Is it possible for us to map the values for providers we know about?

File: packages/start-slicemachine/src/StartSliceMachineProcess.ts
Comment (angeloashmore) [inline packages/start-slicemachine/src/StartSliceMachineProcess.ts:304]: Does Segment do something weird with `@`?

File: packages/start-slicemachine/src/lib/safelyExecute.ts
Comment (angeloashmore) [inline packages/start-slicemachine/src/lib/safelyExecute.ts:1]: This is pedantic, but `safelyExecute` implies the function will execute something. Instead, this functions awaits something that was already executed in a Promise.

How about `safelyAwait` or `safelyTry`?

Or change `operation` to a function that executes the function when `safetyExecute` is called:

```ts
safelyExecute(() => this._sliceMachineManager.simulator.getLocalSliceSimulatorURL())
```

**Review by angeloashmore — APPROVED** [review]
Looks good! I have some small comments, none of which block the PR from being merged. I'll let you decide if the suggestions are necessary.

File: packages/start-slicemachine/src/StartSliceMachineProcess.ts
Comment (angeloashmore) [inline packages/start-slicemachine/src/StartSliceMachineProcess.ts]: Is there a reason we delay tracking to the end of the process? Since it happens in the background and shouldn't affect start up performance significantly, I wonder if we should move this immediately after initializing telemetry and plugins

File: packages/manager/src/managers/git/GitManager.ts
Comment (xrutayisire) [reply packages/manager/src/managers/git/GitManager.ts:475]: I did that at the beginning, but it was less powerful than this. The goal here is really to just get the domain and let the domain tell us what they us. 

Thanks to this and not having a fixed list, it will help to get all possible providers / custom self hosted solution. I would differentiate the tracking compare to the git integration then. WDYT?

I will merge that but we can always change after without a problem!

File: packages/start-slicemachine/src/StartSliceMachineProcess.ts
Comment (xrutayisire) [reply packages/start-slicemachine/src/StartSliceMachineProcess.ts:304]: Yes, I will precise my comment. Segment treat email with * and does this "y***@b***" for example

File: packages/start-slicemachine/src/lib/safelyExecute.ts
Comment (xrutayisire) [reply packages/start-slicemachine/src/lib/safelyExecute.ts:1]: 100% right, I like your second proposal 🙂

File: packages/start-slicemachine/src/StartSliceMachineProcess.ts
Comment (xrutayisire) [reply packages/start-slicemachine/src/StartSliceMachineProcess.ts]: You're right it's even better now!

File: packages/manager/src/managers/git/GitManager.ts
Comment (angeloashmore) [reply packages/manager/src/managers/git/GitManager.ts:475]: When we need to use `detectGitProvider` for other uses, we can revisit. This version works well for tracking, indeed. 🙂

File: packages/start-slicemachine/src/StartSliceMachineProcess.ts
Comment (angeloashmore) [reply packages/start-slicemachine/src/StartSliceMachineProcess.ts:304]: Ah, got it! To anonymize the data, I assume.

**Pull request merged**
